grakn-graph/src/test/java/ai/grakn/graph/internal/EntityTest.java

```diff
@@ -348,4 +330,18 @@ public void testNoKey() throws GraknValidationException {
         expectedException.expect(GraknValidationException.class);
         graknGraph.validateGraph();
     }
+
+    private void checkResourceStructure(ResourceType<?> resourceType, Relation relation, Entity entity){
+        relation.allRolePlayers().entrySet().forEach(entry -> {
+            RoleType roleType = entry.getKey();
+            entry.getValue().forEach(instance -> {
+                if(instance.equals(entity)){
+                    assertEquals(Schema.Resource.HAS_RESOURCE_OWNER.getName(resourceType.getName()), roleType.getName());
+                } else {
+                    assertEquals(Schema.Resource.HAS_RESOURCE_VALUE.getName(resourceType.getName()), roleType.getName());
+                }
+            });
+        });
+    }
```

Comment:
We should check that there are two role types, and that each has exactly one role-player.
This isn't an important issue though, I'm happy to merge without it.